### PR TITLE
fix: introduce new input variable use_default_backup_encryption_key

### PR DESCRIPTION
**Pull request by akocbek**
### Description

Add a new boolean variable called use_default_backup_encryption_key (defaulting to false) which allows consumers to opt in to use the the default ICD key in order to allow them to provision to regions that do not support BYOK backup encryption.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -217,6 +217,12 @@ variable "backup_encryption_key_crn" {
   }
 }
 
+variable "use_default_backup_encryption_key" {
+  type        = bool
+  description = "Set to true to use the default ICD key."
```
Comment (ocofaigh) [inline variables.tf:222]: Set to true to use default ICD randomly generated keys

**Review by ocofaigh — CHANGES_REQUESTED** [review]
The `backup_encryption_key_crn` variable descrption now needs to be updated. Update this sentence:

`If no value is passed here, the value passed for the 'kms_key_crn' variable is used.` -> `If no value is passed here, the value passed for the 'kms_key_crn' variable is used, unless 'use_default_backup_encryption_key' is set to 'true'.`

Also we now need to add validation to ensure that a value cannot be passed for `backup_encryption_key_crn` if `use_default_backup_encryption_key` is set to `true`. This should work (not tested):

```
validate_backup_key = var.backup_encryption_key_crn != null && var.use_default_backup_encryption_key == true ? tobool("When passing a value for 'backup_encryption_key_crn' you cannot set 'use_default_backup_encryption_key' to 'true'") : true
```

**Comment by akocbek** [comment]
@ocofaigh thank you for review.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**